Permanent solution for incorrect UTF-8 in terminal

ReTWi opened the pull request:
Now we have a problem with incorrectly visualized log lines if they include utf8 characters. Executing in the terminal `chcp 65001` will need to be done everytime you run VScode, so i added a permanent solution to this problem for VScode.

ReTWi commented:
Didn't found at the first look a solution for all windows terminals, but this opltions line can be used in other terminals as i can see, just different approches to make it automatically.

davidmarkclements commented:
@ReTWi can we rephrase it to select that, and then provide vscode as an example of that

ReTWi commented:
Found a solution for all terminals in windows using registry (tested all terminals that i know how to open: cmd, powershell and VScode). English is note my native, so please edit if something is wrong. @davidmarkclements

davidmarkclements commented:
@ReTWi amazing work - would you be up for adding a `.reg` file to this PR that a user can simply run to update the registry for them? Then the instructions can simply be "For a permanent solution, download cmd-utf-8-fix.reg file from this repo and run it."

ReTWi commented:
I think i can, just where to place this file to download from? @davidmarkclements

davidmarkclements commented:
just add it to the root of this repo, and add it to .npmignore so it's not installed via npm

davidmarkclements commented:
have you verified this works on power shell as well as cmd?

ReTWi commented:
Yep. Without it i see strange symbols in cmd and powershell (and VScode powershell), but after i add it in this terminals all ok.

davidmarkclements commented on docs/help.md:
```diff
@@ -211,5 +211,6 @@ terminal (both `cmd.exe` and powershell), and as such the output could
 correctly with the use of [`chcp`](https://ss64.com/nt/chcp.html) by
-executing in the terminal `chcp 65001`. This is a known limitation of
-Node.js.
+executing in the terminal `chcp 65001`. As a permanent solution, 
```
```suggestion
executing in the terminal `chcp 65001`. For persistent terminal configuration, the supplied
`windows-cmd-utf8-fix.reg` file can be used to automatically modify the Windows Registry 
to automatically run  `chcp 65001` when a Windows cmd shell or PowerShell is opened.
See the contents of the `windows-cmd-utf8-fix.reg` for details.

```

davidmarkclements commented on docs/help.md:
```diff
@@ -211,5 +211,6 @@ terminal (both `cmd.exe` and powershell), and as such the output could
-executing in the terminal `chcp 65001`. This is a known limitation of
-Node.js.
+executing in the terminal `chcp 65001`. As a permanent solution, 
+you can open windows regedit, go to `[HKEY_LOCAL_MACHINE\Software\Microsoft\Command Processor]`
```
```suggestion

```

davidmarkclements commented on docs/help.md:
```diff
@@ -211,5 +211,6 @@ terminal (both `cmd.exe` and powershell), and as such the output could
-Node.js.
+executing in the terminal `chcp 65001`. As a permanent solution, 
+you can open windows regedit, go to `[HKEY_LOCAL_MACHINE\Software\Microsoft\Command Processor]`
+and add a line with name `Autorun` of type `string value` and data `chcp 65001`.
```
```suggestion

```

davidmarkclements commented:
@jsumners @mcollina can you weigh in on the new approach?

ReTWi commented:
Dash is removed. I don't see your changes in top level files tab, hope this is how is should be.

davidmarkclements commented:
@ReTWi did you click "Commit Suggestion" or just "resolve conversation"? 
gh should generate a commit as well - so it's not worked. If you can't see a way to do it can you copy paste and commit?

ReTWi commented:
Unresolved and commited, looks fine. Sorry for trouble.

jsumners commented on docs/help.md:
```diff
@@ -211,5 +211,8 @@ terminal (both `cmd.exe` and powershell), and as such the output could
 correctly with the use of [`chcp`](https://ss64.com/nt/chcp.html) by
-executing in the terminal `chcp 65001`. This is a known limitation of
-Node.js.
+executing in the terminal `chcp 65001`. For persistent terminal configuration, the supplied
```
Small nit: the line break formatting is out of sync with the rest of the document.

jsumners commented on docs/help.md:
```diff
@@ -211,5 +211,8 @@ terminal (both `cmd.exe` and powershell), and as such the output could
-executing in the terminal `chcp 65001`. This is a known limitation of
-Node.js.
+executing in the terminal `chcp 65001`. For persistent terminal configuration, the supplied
+`windows-cmd-utf8-fix.reg` file can be used to automatically modify the Windows Registry 
```
```suggestion
`windows-cmd-utf8-fix.reg` file can be used to modify the Windows Registry 
```

ReTWi commented:
**Update**. Now i don't think that changing registy is a god idea. After i changed it i had unexprected problems with PUSH command using github desktop with any console, that i fixed only by deleting this line in registry. There maybe other issues that we don't know, so i suggest to rollback to my first approch with applying chcp 65001 only in editor where you need to see logs. @davidmarkclements

davidmarkclements commented:
@ReTWi 🤣 

can you verify whether it's the command chcp command itself that causes problems with git push or whether it's the autorun config that causes issues

because if it's the command itself then git push is screwed either way.. 

I think there's another option:  create a module called pino-win-utf8, used like so

```sh
node -r pino-win-utf8 app.js
```

What the module does is run the chpc command on init, and then resets it back to the prior setting on program exit.

davidmarkclements commented:
I mean if that is the case, the fact that a terminal configuration can mess up git authentication would be a windows git bug

ReTWi commented:
@davidmarkclements 
Ok, to make it clear, if i add this line to registry, i have a problem to PUSH using Github Desktop.

At the same time a have no problem to push using GIT Bush. So problem is with Github Desktop.
If this autorun using registry somehow broking Github Desktop, than it can broke other things, so i vote fore local solutions.

Here is my gitbash push:
```
Active code page: 65001
acer@PredatorRTW MINGW64 /d/apps/app (master)
$ git push origin master
Enumerating objects: 28, done.
Counting objects: 100% (28/28), done.
Delta compression using up to 8 threads
Compressing objects: 100% (15/15), done.
Writing objects: 100% (15/15), 1.51 KiB | 775.00 KiB/s, done.
Total 15 (delta 9), reused 0 (delta 0)
remote: Resolving deltas: 100% (9/9), completed with 9 local objects.
To https://github.com/ReTWi/app.git
   0d1c4eb..72f6bf2  master -> master

acer@PredatorRTW MINGW64 /d/apps/app (master)
$
```
